docs/pre-release/argo/canary-rollouts-with-argo.md

```diff
@@ -0,0 +1,262 @@
+
+```
+curl -Lk "https://$AMBASSADOR_LB_ENDPOINT/echo/"
+VERSION 1
+```
+
+Congratulations, you've deployed your first Rollout service successfully!
+
+## 6. Rollout a new version 
+
+It's time to rollout a new version of the service. Update the echo container in the `rollout.yaml` to display "VERSION 2":
+
+```
+apiVersion: argoproj.io/v1alpha1
+kind: Rollout
+metadata:
+  name: echo-rollout
+spec:
+  selector:
+    matchLabels:
+      app: echo
+  template:
+    metadata:
+      labels:
+        app: echo
+    spec:
+      containers:
+        - image: hashicorp/http-echo
+          args:
+            - "-text=VERSION 2"
+            - -listen=:8080
+          imagePullPolicy: Always
+          name: echo-v1
+          ports:
+            - containerPort: 8080
+  strategy:
+    canary:
+      stableService: echo-stable
+      canaryService: echo-canary
+      trafficRouting:
+        ambassador:
+          mapping: echo
+      steps:
+      - setWeight: 30
+      - pause: {duration: 30s}
+      - setWeight: 60
+      - pause: {duration: 30s}
+      - setWeight: 100
+      - pause: {duration: 10}
+```
+
+Apply the rollout to the cluster by typing `kubectl apply -f rollout.yaml`. This will rollout a version 2 of the service by routing 30% of traffic to the service for 30 seconds, followed by 60% of traffic for another 30 seconds.
+
+You can monitor the status of your rollout at the command line:
+
+```
+kubectl argo rollouts get rollout echo-rollout --watch
+```
+
+Will display an output similar to the following:
+
+```
+Name:            echo-rollout
+Namespace:       default
+Status:          ॥ Paused
+Message:         CanaryPauseStep
+Strategy:        Canary
+  Step:          1/6
+  SetWeight:     30
+  ActualWeight:  30
+Images:          hashicorp/http-echo (canary, stable)
+Replicas:
+  Desired:       1
+  Current:       2
+  Updated:       1
+  Ready:         2
+  Available:     2
+
+NAME                                      KIND        STATUS        AGE    INFO
+⟳ echo-rollout                            Rollout     ॥ Paused      2d21h
+├──# revision:3
+│  └──⧉ echo-rollout-64fb847897           ReplicaSet  ✔ Healthy     2s     canary
+│     └──□ echo-rollout-64fb847897-49sg6  Pod         ✔ Running     2s     ready:1/1
+├──# revision:2
+│  └──⧉ echo-rollout-578bfdb4b8           ReplicaSet  ✔ Healthy     3h5m   stable
+│     └──□ echo-rollout-578bfdb4b8-86z6n  Pod         ✔ Running     3h5m   ready:1/1
+└──# revision:1
+   └──⧉ echo-rollout-948d9c9f9            ReplicaSet  • ScaledDown  2d21h
+```
+
+In your other terminal window, you can verify that the canary is progressing appropriately by sending requests in a loop:
+
+```
+while true; do curl -k https://$AMBASSAOR_LB_ENDPOINT/echo/; sleep 0.2; done
+```
+
+will display a running list of responses from the service that will gradually transition from VERSION 1 strings to VERSION 2 strings.
+
+Congratulations! You've successfully integrated Argo Rollouts with Edge Stack.
```

Comment:
add "Read more" section at the end with links to Argo Rollouts, Edge Stack, etc!?